File: src/main/java/com/github/dkuzmenkov17/todo/dao/ToDoRepository.java
package com.github.dkuzmenkov17.todo.dao;

import com.github.dkuzmenkov17.todo.dao.jpa.ToDoJpaEntity;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface ToDoRepository extends PagingAndSortingRepository<ToDoJpaEntity, Long>,
        QuerydslPredicateExecutor<ToDoJpaEntity> {
}

File: src/main/kotlin/com/github/dkuzmenkov17/todo/dao/ToDoRepository.kt
package com.github.dkuzmenkov17.todo.dao

import com.github.dkuzmenkov17.todo.dao.jpa.ToDoEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.PagingAndSortingRepository

interface ToDoRepository : JpaRepository<ToDoEntity, Long>